src/Models/AzureComponentConfig.ts

```diff
@@ -107,11 +129,10 @@ export class AzureConfigFileHandler {
     }
   }
 
-  getComponentIndexById(id: string) {
+  async getComponentIndexById(id: string) {
     try {
-      const azureConfigs =
-          JSON.parse(fs.readFileSync(this.configFilePath, 'utf8')) as
-          AzureConfigs;
+      const azureConfigContent = await FileUtility.readFileInLocal(this.configFilePath, 'utf8');

```

Comment:
no idea why use async here. any benefit?

Reply:
The reason is that API we used from fs-plus is **sync**, but API from cube is **async**. If we combine the two API into one, we have to use **async** since we are not certain which API will be used.